ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -99,7 +101,11 @@
         if (replicaDcName.equals(datacenterName)) {
           replicaPool.addFirst(replicaId);
         } else if (crossColoEnabled) {
-          replicaPool.addLast(replicaId);
+          if (replicaDcName.equals(crossColoPreferredDc)) {
```

Comment:
Do you think it is better to extract the id out and compare them rather than the other way? That way, you do not need to add a new method to the `clusterMap` interface.

Reply:
Current change is using a hashMap to know every replica's DC name and re-ordering them base on this. Therefore a method in clusterMap is added. 

What is the other way? get DC name by something like replica.getDCname()?

Reply:
There isn't, you can ignore the comment. I thought there was a method to get the id given a name. Partly because the method names are a bit confusing to me. `getDatacenterById()` and `getDatacenterName()` both look up using an id.